packages/aws-rfdk/lib/deadline/lib/version.ts

```diff
@@ -72,6 +72,106 @@ abstract class VersionQueryBase extends Construct implements IVersion {
   }
 }
 
+/**
+  * This class is reposonsible to do basic operations on version format.
+  */
+export class Version {
+
+  /**
+   * This method parses the input string and returns the version object.
+   *
+   * @param version version string to parse
+   */
+  public static parse(version: string): Version {
+    if (!Version.validateVersionFormat(version)) {
+      throw new TypeError(`Invalid version format. Expected format 'a.b.c.d', found '${version}'`);
+    }
+
+    return new Version(version.split('.').map(x => parseInt(x)));
+  }
+
+  /**
+   * This method validates the given string for a sequence '.' separated numbers.
+   *
+   * @param version the string to be validated.
+   *
+   * @returns true if the format is correct, else false.
+   */
+  private static validateVersionFormat(version: string): boolean {
+    /**
+     * Regex: ^\d+(?:\.\d+){3}$
+     * Matches a sequence of '.' separated numbers with exactly 4 digits.
+     * - ^ asserts position at start of a line.
+     * - \d+ Matches one or more digits.
+     * - (?:\.\d+) Matches a dot and the following one or more digits.
+     * - {3} Matches previous pattern exactly 3 times.
+     * - $ asserts position at the end of a line
+     */
+    if (version.match(/^\d+(?:\.\d+){3}$/)) {
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Numeric components of version.
+   */
+  public readonly components: number[];
+
+  constructor(components: number[]) {
+    this.components = components;
```

Comment:
Should we add some input validation here as well and the corresponding tests? We are accepting arbitrary number of components and negative numbers.

Reply:
Added.